--- a/config/apparmor/profiles/lxc-default	2014-05-26 15:40:45.000000000 +0700
+++ b/config/apparmor/profiles/lxc-default	2014-05-26 15:34:25.000000000 +0700
@@ -1,11 +1,28 @@
 # Do not load this file.  Rather, load /etc/apparmor.d/lxc-containers, which
 # will source all profiles under /etc/apparmor.d/lxc
 
-profile lxc-container-default flags=(attach_disconnected,mediate_deleted) {
+profile lxc-container flags=(attach_disconnected,mediate_deleted) {
   #include <abstractions/lxc/container-base>
 
   # the container may never be allowed to mount devpts.  If it does, it
   # will remount the host's devpts.  We could allow it to do it with
   # the newinstance option (but, right now, we don't).
   deny mount fstype=devpts,
+  
+  # recursive mounts (with -R flag)
+  mount options=(rw, rbind),
+  mount options=(rw, bind),
+  
+  # cgroup mounting under container
+  mount fstype=cgroup -> /sys/fs/cgroup/{**,},
+  mount options=(ro, nosuid, nodev, noexec, remount, strictatime) -> /sys/fs/cgroup/,
+  
+  # permit hugetablefs
+  mount fstype=hugetlbfs -> /dev/hugepages/,
+  
+  # permit configfs
+  mount fstype=configfs -> /sys/kernel/config/,
+  
+  # deny udevd or other to make root rslave
+  deny mount options=(rw, rslave) -> /,
 }
